Restructure installation and add configuration docs (#3823)
kyma-project/kyma: docs/kyma/docs/05-01-overview.md

---
title: Overview
type: Configuration
---

## Default settings

> During the installation process, the Installer applies onto a cluster or Minikube all components defined in the `installer-cr-cluster.yaml.tpl` and `installer-cr.yaml.tpl` templates with their configuration defined in `values.yaml` files. It also imports the configuration settings defined in the `installer-config-cluster.yaml.tpl` and `installer-config-local.yaml.tpl` templates located under the `installation/resources` subfolder.

Reviewer: `installer-cr-cluster.yaml.tpl` or `installer-cr.yaml.tpl`

`imports the configuration settings` = `imports the configuration overrides`?

Author: Changed that :)

>**CAUTION:** Both `values.yaml` and the configuration files contain production settings that are tested and recommended. Note that you modify them on your own risk.

## Installation configuration

Before you start the Kyma installation process, you can customize the default settings in the installation and configuration files.

### Components

During the installation, the Installer creates the Kyma Installer image that contains all component charts located in the `kyma/resources` folder. Both cluster and local installation files contain a full list of these component names and their Namespaces.
Components that are not an integral part of the default Kyma Lite package, are preceded with hashtags (#) which means that the Installer skips them during the installation process.   
You can customize the component installation files by:
- Removing hashtags in front of the component entries to enable a component installation
- Adding hashtags to disable a component installation

> - Adding new components to the list along with their chart definition in the `kyma/resources` folder.

Reviewer: here should also be a link to doc about building your own installer image, as this is needed when adding new chart

Author: Ok, I added it.

---
title: Overview
type: Configuration
---

## Default settings

During the installation process, the Installer applies onto a cluster or Minikube all components defined in the `installer-cr-cluster.yaml.tpl` or `installer-cr.yaml.tpl` templates with their configuration defined in `values.yaml` files. It also imports the configuration overrides defined in the `installer-config-cluster.yaml.tpl` and `installer-config-local.yaml.tpl` templates located under the `installation/resources` subfolder.

> **NOTE:** The installation and configuration templates serve as the basis for creating the installation (`kyma-installer-local.yaml` and `kyma-installer-cluster.yaml`) and configuration (`kyma-config-local.yaml` and `kyma-config-cluster.yaml`) release artefacts.

> Both configuration files contain pre-defined overrides in the form of Secrets and ConfigMaps that the Installer uses during the installation to replace default values specified in `values.yaml` files or to provide required configuration values. While the local configuration file contains hardcoded override values, the cluster configuration file is based on placeholders that are replaced with real values during the installation process.

Reviewer 2: ```suggestion
Both configuration files contain pre-defined overrides in the form of Secrets and ConfigMaps that the Installer uses during the installation to replace default values specified in `values.yaml` files or to provide required configuration values. While the local configuration file contains hardcoded override values, the cluster configuration file is based on placeholders replaced with real values during installation.
```

Author: but `during the installation`

You can add multiple Secrets and ConfigMaps for one component. If you duplicate the same parameter in a few Secrets or ConfigMaps for a given component but give it different values, the Installer accepts the last value in the file.

>**CAUTION:** Both `values.yaml` and the configuration files contain production settings that are tested and recommended. Note that you modify them on your own risk.

## Installation configuration

Before you start the Kyma installation process, you can customize the default settings in the installation and configuration files.

### Components

One of the installation artifacts is the Kyma Installer image that contains all component charts located in the `kyma/resources` folder. Both cluster and local installation files contain a full list of these component names and their Namespaces.
Components that are not an integral part of the default Kyma Lite package, are preceded with hashtags (#) which means that the Installer skips them during the installation process.   

> You can customize the component installation files by:

Reviewer 2: ```suggestion
You can customize component installation files by:
```

- Removing hashtags in front of the component entries to enable a component installation.
- Adding hashtags to disable a component installation.
- Adding new components to the list along with their chart definition in the `kyma/resources` folder. In that case you must create your own [Kyma Installer image](#configuration-use-your-own-kyma-installer-image) as you are adding new chart configuration.

For more details on custom component installation, see [this](#configuration-custom-component-installation) document.

### Overrides

The configuration files provide production-ready settings for the Installer. These include ConfigMaps and Secrets with the overrides for the values hardcoded in the `values.yaml` files of component charts.

> You can modify local settings, such as memory limits for a given resource, by changing values in the existing ConfigMaps and Secrets or adding new ones. Every ConfigMap and Secret that the Installer reads must contain an `installer: overrides` label and a `component: {component-name}` label if it refers to a specific component. Add the `kyma-project.io/installation: ""` label to all non-Kubernetes resources to delete or search for them easier.

Reviewer: you have already 3 label, it make sense to split them into 3 bullet points

Every ConfigMap and Secret that the Installer reads must contain the following labels:
* `installer: overrides`
* `component: {COMPONENT_NAME}` if resource refers to a specific component
* `kyma-project.io/installation: ""` to all non-Kubernetes resources to delete or search for them easier.

btw, tbh I don't understand `to delete or search for them easier.`

Author: I created the bullet points and improved the explanation of the `kyma-project.io/installation: ""` label

[Read more](#configuration-helm-overrides-for-kyma-installation) about the types of overrides and the rules for creating them.

## Runtime configuration

Upon installing Kyma, you can still modify the installation artifacts. However, for these changes to take effect, you must trigger the update process:  

```
kubectl label installation/kyma-installation action=install
```

>**NOTE:** You cannot uninstall a Kyma component that is already installed. If you remove it from any of the installation files or add hashtags in front of its `name` and `namespace` entries, you only disable its further updates.

> Apart from modifying the installation artifacts, you can additionally configure the installed components and their behavior in the runtime using ConfigMaps and Secrets.

Reviewer 2: ```suggestion
Apart from modifying the installation artifacts, you can use ConfigMaps and Secrets to configure the installed components and their behavior in the runtime.
```

> For an example of such a runtime configuration, see [Helm Broker configuration](/components/helm-broker/#configuration-configuration) in which you add links the bundle repositories to a ConfigMap and label it with the `helm-broker-repo=true` label for the Helm Broker to expose additional Service Classes in the Service Catalog.

Reviewer 2: ` in which you add links the bundle repositories to a ConfigMap and label it with the `helm-broker-repo=true` label`

should it be ` in which you add links to the bundle repositories to a ConfigMap`?  (missing preposition)
